### PR TITLE
Connector get resources

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

Connector exposes a logic through `BATCH_RETRIEVE_RESOURCE_TITLE_BY_TYPE` that is used in front to store the selected datasources of an assistant with route `/connectors/:connector_id/resources/titles` -> from a list of internal ids it retrieves their document titles. 

Since we'll want to display the tree of selected datasource (part 3 of https://github.com/dust-tt/decisions/issues/164), we need connector to expose directly the `ConnectorNodes` instead of just the title of the resources. 

On this PR I'm adding a `BATCH_RETRIEVE_RESOURCE_BY_TYPE` logic with a new route `/connectors/:connector_id/resources`. 

Implementation of the function is done only for Intercom and Notion, I'll put the other connectors in another PR. 

This could be merged as is as it's not called yet. 

Once this is complete and front is calling the resource route over the resource title one, I'll open a cleanup PR to remove the title logic. 

The front PR using this is there: https://github.com/dust-tt/dust/pull/4065/files but still in draft as I wanted to make sure this one gets checked first 🙇🏻‍♀️ 

## Risk

Low cause not called yet. 

## Deploy Plan

Nothing special.

### CODE REVIEW

File: connectors/src/api_server.ts [hunk 1]
```diff
@@ -106,6 +107,11 @@ export function startServer(port: number) {
     "/connectors/:connector_id/resources/titles",
     getResourcesTitlesAPIHandler
   );
+  app.post(
+    // must be POST because of body
+    "/connectors/:connector_id/resources",
```
Comment (fontanierh) [inline connectors/src/api_server.ts:112]: as per IRL: nodes would be a better name since we use `resource` for our "models" now

File: connectors/src/connectors/intercom/lib/help_center_permissions.ts [hunk 2]
```diff
@@ -381,7 +381,7 @@ export async function retrieveIntercomHelpCentersPermissions({
         expandable: true,
         permission: helpCenter.permission,
         dustDocumentId: null,
-        lastUpdatedAt: null,
+        lastUpdatedAt: helpCenter.updatedAt.getTime(),
```
Comment (fontanierh) [inline connectors/src/connectors/intercom/lib/help_center_permissions.ts:384]: seems unrelated, is it on purpose ?

**Review by fontanierh — APPROVED** [review]
Looks great (modulo small renaming) !

File: connectors/src/connectors/intercom/lib/help_center_permissions.ts (hunk 2 above)
Comment (PopDaph) [reply connectors/src/connectors/intercom/lib/help_center_permissions.ts:384]: Yes totally unrelated I should have added a comment to make it clear 😬

**Review by spolu — COMMENTED** [review]
LGTM

As per IRL let's standardize on `ContentNode` / `connectors/:connector_id/content_nodes`
